### ISSUE TITLE
Match overload based on usertype

### ISSUE THREAD

**Issue by gotschmarcel-ni** [ThePhD/sol2#1306]
I have two user types registered and I want to pass them to C++ and match an overload based on their type. Is that possible?

This is an example, but it doesn't work. Always goes to the first overload.

https://godbolt.org/z/8cW9ovWY7

**Comment by ThePhD** [comment]
Ah, on the type.

You can't do that right now because I didn't write a checker for this. This just grabs the table that repesents the usertype, not specifically _that_ usertype. I could probably add a discriminator for this case in particular.

**Comment by ThePhD** [comment]
IF you're looking just for the value, that should work fine:

https://godbolt.org/z/vMnobvEzo

**Comment by gotschmarcel-ni** [comment]
Yeah the value works fine, but I'm actually interested in the user type itself. Would be nice if that would be possible.
